libkbfs/block_disk_store.go

```diff
@@ -230,38 +240,72 @@ func (s *blockDiskStore) getData(id BlockID) (
 // All functions below are public functions.
 
 func (s *blockDiskStore) hasAnyRef(id BlockID) (bool, error) {
-	refInfo, err := s.getRefInfo(id)
+	info, err := s.getInfo(id)
 	if err != nil {
 		return false, err
 	}
 
-	return len(refInfo.Refs) > 0, nil
+	return len(info.Refs) > 0, nil
 }
 
 func (s *blockDiskStore) hasNonArchivedRef(id BlockID) (bool, error) {
-	refInfo, err := s.getRefInfo(id)
+	info, err := s.getInfo(id)
 	if err != nil {
 		return false, err
 	}
 
-	return refInfo.Refs.hasNonArchivedRef(), nil
+	return info.Refs.hasNonArchivedRef(), nil
 }
 
 func (s *blockDiskStore) hasContext(id BlockID, context BlockContext) (
 	bool, error) {
-	refInfo, err := s.getRefInfo(id)
+	info, err := s.getInfo(id)
 	if err != nil {
 		return false, err
 	}
 
-	return refInfo.Refs.checkExists(context)
+	return info.Refs.checkExists(context)
 }
 
 func (s *blockDiskStore) hasData(id BlockID) error {
 	_, err := ioutil.Stat(s.dataPath(id))
 	return err
 }
 
+type errFlushedButPresentData struct{}
+
+func (e errFlushedButPresentData) Error() string {
+	return "Data is flushed but present"
+}
+
+func (s *blockDiskStore) isUnflushed(id BlockID) error {
```

Comment:
I'm worried about swallowing errors that aren't `errFlushedButPresentData`  or `isNotExist`. Can you make this function (and `hasData`, too, I guess) like `hasAnyRef`, i.e. return a `bool, error`?